src/utils/useDelayed.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { useState, useEffect } from 'react'
+
+type Props = {
+  children: React.ReactNode
+  waitBeforeShow?: number
+}
+
+export const Delayed = ({ children, waitBeforeShow = 1500 }: Props) => {
```

Comment:
Nitpick: use `PropsWithChildren`

Example: https://github.com/cowprotocol/cowswap/blob/10d798be0f843861c9c5f5380d299c23a96474cf/src/legacy/components/AnimatedDropdown/index.tsx#L8

```suggestion
import { useState, useEffect, PropsWithChildren } from 'react'

type Props = {
  waitBeforeShow?: number
}

export const Delayed = ({ children, waitBeforeShow = 1500 }: PropsWithChildren<Props>) => {
```